presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -377,19 +398,113 @@ public TableStatistics getTableStatistics(ConnectorTransactionHandle transaction
 
         long numRowsSum = partitionRowNums.stream().mapToLong(a -> a).sum();
         long partitionsWithStatsCount = partitionRowNums.size();
-        long allPartitionsCount = hivePartitions.size();
+        long allPartitionsCount = partitionStatisticsMap.size();
+        return new Estimate(1.0 * numRowsSum / partitionsWithStatsCount * allPartitionsCount);
+    }
+
+    private Estimate calculateDistinctValuesCount(Map<String, PartitionStatistics> partitionStatisticsMap, String column)
+    {
+        return getColumnStatisticsValue(
+                partitionStatisticsMap,
+                column,
+                columnStatistics -> {
+                    switch (columnStatistics.getType()) {
+                        case LONG:
+                            return OptionalDouble.of(columnStatistics.getLongColumnStatistics().getDistinctValuesCount());
```

Comment:
can't you make column statistics class extend from some `AbstractColumnStatistics` and have those methods there?